content/en/blog/2024/otel-generative-ai/index.md

```diff
@@ -0,0 +1,163 @@
+Two primary assets are in development to make this possible: **Semantic
+Conventions** and an **Instrumentation Library**.
+
+[**Semantic Conventions**](https://opentelemetry.io/docs/concepts/semantic-conventions/)
+establish standardized guidelines for how telemetry data is structured and
+collected across platforms, defining inputs, outputs, and operational details.
+For generative AI, these conventions streamline monitoring, troubleshooting, and
+optimizing AI models by standardizing attributes such as model parameters,
+response metadata, and token usage. This consistency supports better
+observability across tools, environments, and APIs, helping organizations track
+performance, cost, and safety with ease.
+
+The
+[**Instrumentation Library**](https://opentelemetry.io/docs/specs/otel/overview/#instrumentation-libraries)
+is being developed within the
+[OpenTelemetry Python Contrib](https://github.com/open-telemetry/opentelemetry-python-contrib)
+project to automate telemetry collection for generative AI applications. The
+first release is a Python library for instrumenting OpenAI client calls, given
+Python's widespread use in AI development and the popularity of OpenAI. Designed
+to integrate seamlessly with OpenAI's API, this library captures spans and
+events, gathering essential data like model inputs, response metadata, and token
+usage in a structured format.
+
+## Key Signals for Generative AI
+
+The
+[Semantic Conventions for Generative AI](https://github.com/open-telemetry/semantic-conventions/tree/v1.28.0/docs/gen-ai)
+focus on capturing insights into AI model behavior through three primary
+signals: [Traces](https://opentelemetry.io/docs/concepts/signals/traces/),
+[Metrics](https://opentelemetry.io/docs/concepts/signals/metrics/), and
+[Events](https://opentelemetry.io/docs/specs/otel/logs/event-api/).
+
+Together, these signals provide a comprehensive monitoring framework, enabling
+better cost management, performance tuning, and request tracing.
+
+### Traces: Tracing Model Interactions
+
+Traces track each model interaction’s lifecycle, covering input parameters (for
+example, temperature, top_p) and response details like token count or errors.
+They provide visibility into each request, aiding in identifying bottlenecks and
+analyzing the impact of settings on model output.
+
+### Metrics: Monitoring Usage and Performance
+
+Metrics aggregate high-level indicators like request volume, latency, and token
+counts, essential for managing costs and performance. This data is particularly
+critical for API-dependent AI applications with rate limits and cost
+considerations.
+
+### Events: Capturing Detailed Interactions
+
+Events log detailed moments during model execution, such as user prompts and
+model responses, providing a granular view of model interactions. These insights
+are invaluable for debugging and optimizing AI applications where unexpected
+behaviors may arise.
+
+> Note that we decided to use the newer Events API
```

Comment:
@svrnm I've added this blockquote as a note on our use of the Events API. Does this resolve the issue you are raising around logs vs events?

Reply:
Yes, that's perfect 🙌